cirq-google/cirq_google/engine/engine_client.py

```diff
@@ -1090,3 +1106,36 @@ def _date_or_time_to_filter_expr(param_name: str, param: Union[datetime.datetime
         f"type {type(param)}. Supported types: datetime.datetime and"
         f"datetime.date"
     )
+
+
+def _construct_device_config_key(
+    run_name: Optional[str], config_alias: Optional[str]
+) -> Optional[quantum.DeviceConfigKey]:
+    if run_name and not config_alias:
+        raise ValueError("Must specify `config_alias` if `run_name` is non-empty")
```

Comment:
As far as I know, there is always a `current` automation run. This is not the case for config_aliases, as they can vary between automation runs. For example, many automation runs don't  have the canonical `GRID_TO_MANTAIN` (stored in keys.GRID_MAINTENANCE_ALIAS in the keys file of the cirq_server). 
e.g. the BLACK processor doesn't have it https://qh-web.corp.google.com/registry/Automation/SYC01_10A_BLACK/current/_configs/)

However, in cirq_server there is a default pair (run_name, config_alias) that together make up the default configuration, that you can find here (https://qh-web.corp.google.com/registry/Servers/cirq_server/runners/SYC01_10A_BLACK/). This is used if we don't specify any of the run_name and config_alias fields. However I can see how this could be confusing, and maybe I should not allow DeviceConfigKey("", "")  either.

This is why specyfing a configuration with empty run_name could always default to current, but not the other way around